src/components/common/Calendar/Calendar.jsx

```diff
@@ -174,37 +196,54 @@ function Calendar({
 
 // Static functions
 Calendar.daysOfMonth = (month) => {
-  const iterator = new Date(month.getTime());
-  const days = [];
-
-  while (iterator.getMonth() === month.getMonth()) {
-    days.push(Calendar.copy(iterator));
-    iterator.setDate(iterator.getDate() + 1);
+  try {
+    const iterator = new Date(month.getTime());
+    const days = [];
+    while (iterator.getMonth() === month.getMonth()) {
+      days.push(Calendar.copy(iterator));
+      iterator.setDate(iterator.getDate() + 1);
+    }
+    return days;
+  } catch {
+    return [];
   }
-
-  return days;
 };
 
-Calendar.isSameDay = (a, b) => a.getFullYear() === b.getFullYear() &&
+Calendar.isSameDay = (a, b) => {
+  try {
+    a.getFullYear() === b.getFullYear() &&
     a.getMonth() === b.getMonth() &&
     a.getDate() === b.getDate();
-
+  } catch {
+    return false;
+  }
+};
 /**
  * set the date component to 1
  * @param day - remains unchanged
  * @returns {Date}
  */
 Calendar.toStartOfMonth = (day) => {
-  const newDay = Calendar.copy(day);
-  newDay.setDate(1);
+  try {
+    const newDay = Calendar.copy(day);
+    newDay.setDate(1);
   return newDay;
+  } catch {
+    return "1"
+  }
```

Comment:
```suggestion
  try {
    const newDay = Calendar.copy(day);
    newDay.setDate(1);
    return newDay;
  } catch {
    return '1';
  }
```

The linting expects single quotes here